test/Microsoft.AspNetCore.Dispatcher.FunctionalTest/ApiAppStartup.cs

```diff
@@ -62,15 +62,25 @@ public void ConfigureDispatcher(DispatcherOptions options)
             {
                 Endpoints =
                 {
-                    new TemplateEndpoint("api/products", Products_Get),
+                    new TemplateEndpoint("api/products", new { controller = "Products", action = "Get", }, "GET", Products_Get),
+                    new TemplateEndpoint("api/products", new { controller = "Products", action = "Post", }, "POST", Products_Post),
+                    new TemplateEndpoint("api/products", new { controller = "Products", action = "Put", }, "PUT", Products_Put),
                 },
```

Comment:
@rynowak I tried to add endpoints with a different route "api/products/{id}" but that throws an `IndexOutofBoundsException` here https://github.com/aspnet/Routing/blob/dev/src/Microsoft.AspNetCore.Routing/Dispatcher/TreeDispatcher.cs#L170 because it has two `entries` and just one element in `trees`

Reply:
Yeah I just found that bug here too 😆  The correct line should be
```
var tree = trees[entry.Order];
```

Reply:
Go ahead and fix it. This is why we're adding more tests